compat/v1/compat.go

```diff
@@ -0,0 +1,323 @@
+package v1
+
+import (
+	"fmt"
+
+	"github.com/renproject/darknode/engine"
+	"github.com/renproject/multichain"
+	"github.com/renproject/pack"
+)
+
+type StateResponse struct {
+	Bitcoin     UTXOState    `json:"Bitcoin,omitempty"`
+	Bitcoincash UTXOState    `json:"BitcoinCash,omitempty"`
+	Digibyte    UTXOState    `json:"DigiByte,omitempty"`
+	Dogecoin    UTXOState    `json:"Dogecoin,omitempty"`
+	Filecoin    AccountState `json:"Filecoin,omitempty"`
+	Terra       AccountState `json:"Terra,omitempty"`
+	Zcash       UTXOState    `json:"Zcash,omitempty"`
+}
+type Outpoint struct {
+	Hash  string `json:"hash"`
+	Index string `json:"index"`
+}
+type Output struct {
+	Outpoint     Outpoint `json:"outpoint"`
+	Pubkeyscript string   `json:"pubKeyScript"`
+	Value        string   `json:"value"`
+}
+type UTXOState struct {
+	Address           string `json:"address"`
+	Dust              string `json:"dust"`
+	Gascap            string `json:"gasCap"`
+	Gaslimit          string `json:"gasLimit"`
+	Gasprice          string `json:"gasPrice"`
+	Latestchainhash   string `json:"latestChainHash"`
+	Latestchainheight string `json:"latestChainHeight"`
+	Minimumamount     string `json:"minimumAmount"`
+	Output            Output `json:"output"`
+	Pubkey            string `json:"pubKey"`
+}
+type Gnonces struct {
+	Address string `json:"address"`
+	Nonce   string `json:"nonce"`
+}
+type AccountState struct {
+	Address           string    `json:"address"`
+	Gascap            string    `json:"gasCap"`
+	Gaslimit          string    `json:"gasLimit"`
+	Gasprice          string    `json:"gasPrice"`
+	Gnonces           []Gnonces `json:"gnonces"`
+	Latestchainhash   string    `json:"latestChainHash"`
+	Latestchainheight string    `json:"latestChainHeight"`
+	Minimumamount     string    `json:"minimumAmount"`
+	Nonce             string    `json:"nonce"`
+	Pubkey            string    `json:"pubKey"`
+}
+
+func QueryStateResponseFromState(state map[string]engine.XState) (StateResponse, error) {
+	stateResponse := StateResponse{}
+
+	bitcoinS, ok := state[string(multichain.Bitcoin.NativeAsset())]
+	if ok {
+		// do we want to log that we are missing bitcoin
+		// return StateResponse{},
+		// 	fmt.Errorf("Missing Bitcoin State")
```

Comment:
Is this comment still relevant? Seems we're in the "BTC state does exist" case here.

Reply:
Yeah, wanted to ask; as ren-js v2 expects all of these assets to be present, do we want to log errors here (even though valid lightnode/darknode configs will have these responses missing)

Reply:
Yeah, might be better to return an error than a half-populated state. Both cases are unlikely, but worth considering how RenJS would respond in both cases though and implement the least problematic option.

Reply:
Definitely don't want to return an error because it makes testing the lightnode more difficult (always need to have all chains enabled).

Was thinking more around the lines of just logging so that we can debug it easier

Reply:
Decided not to go with logging here; as the behaviour is the same with v2 if the lightnode isn't set up to handle the selector